### PR TITLE
#457 - inactive user cleanup

### LINKED ISSUE #457: Make clean-up with inactive user
So, we have some little inconsistency with inactive employees. 
For example:
- on calendar view I don't see inactive user
- but on vacation requests list I can see his vacation requests (but on users filter inactive users are not shown)
- on vacation limits page I also see inactive user limit

So, after speaking a bit with Kamila, we think that we should by default hide all information about inactive users, but we can make some checkbox for example "Show inactive users vacation requests" and on vacation requests page we can see also inactive user requests, but records have gray background or something. 

And we should change toast text, when admin click on user and click option "Zablokuj". Now we have "Usunięto użytkownika", but this is not entirely true, because all his information is in database. We can change it to "Dezaktywowano użytkownika" or "Zablokowano użytkownika".

### PR DESCRIPTION
**Pull request by kamilpiech97**
Should close #457.
Added gray background color for blocked users if user wanna see them on index. Added on indexes checkbox for switching visibility of inactive users.

Preview:
image

### CODE REVIEW

**Review by krzysztofrewak — COMMENTED** [review]
Let's talk tomorrow about these filters. Please remind me about it.

**Review by EwelinaSkrzypacz — COMMENTED** [review]
How about CVs, devices and benefits? 
And on vacation limit page I see always the inactive user.

**Review by EwelinaSkrzypacz — COMMENTED** [review]
1. As an admin, when I go to page with equipment I see error:

image

2. On CV page, inactive users should be marked somehow - like gray background or so
3. The same thing on page `benefits-reports/1`

3. ON users page, inactive users have red background but on other pages gray. Can we unify it somehow?

**Comment by kamilpiech97** [comment]
> 1. As an admin, when I go to page with equipment I see error:
> 
> image
> 
> 2. On CV page, inactive users should be marked somehow - like gray background or so
> 3. The same thing on page `benefits-reports/1`
> 4. ON users page, inactive users have red background but on other pages gray. Can we unify it somehow?

1, 2, 3, 4 fixed

**Review by EwelinaSkrzypacz — COMMENTED** [review]
On benefits report page still we don't have marked inactive users (at least I don't have)


And I still have problem with equipment

**Review by EwelinaSkrzypacz — APPROVED** [review]

File: app/Http/Controllers/EquipmentController.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public function index(Request $request): RedirectResponse|Response
         $searchQuery = $request->query("search");
 
         $equipmentItems = EquipmentItem::query()
-            ->with("assignee")
+            ->with(["assignee" => fn($query) => $query->withTrashed()])
```
Comment (krzysztofrewak) [inline app/Http/Controllers/EquipmentController.php:36]: Please type these.

File: app/Http/Controllers/EquipmentController.php [hunk 2]
```diff
@@ -49,6 +49,7 @@ public function index(Request $request): RedirectResponse|Response
             ->withQueryString();
 
         $users = User::query()
+            ->withTrashed()
```
Comment (krzysztofrewak) [inline app/Http/Controllers/EquipmentController.php:52]:
```suggestion
            ->withTrashed($request->user()->hasPermissionTo("showInactiveUsers"))
```

Hm?

File: app/Http/Controllers/ResumeController.php [hunk 3]
```diff
@@ -18,12 +20,14 @@
...
 {
-    public function index(): Response
+    public function index(Request $request): Response
     {
         $this->authorize("manageResumes");
 
         $resumes = Resume::query()
             ->with("user")
+            ->whereRelation("user", fn(Builder $query): Builder => $query->withTrashed($request->user()->hasPermissionTo("showInactiveUsers")))
+            ->orWhere("user_id", null)
```
Comment (krzysztofrewak) [inline app/Http/Controllers/ResumeController.php:30]:
```suggestion
            ->orWhereNull("user_id")
```

File: app/Http/Controllers/VacationLimitController.php [hunk 4]
```diff
@@ -25,6 +27,7 @@ public function edit(YearPeriodRetriever $yearPeriodRetriever, UserVacationStats
 
         $limits = $yearPeriod
             ->vacationLimits()
+            ->whereRelation("user", fn(Builder $query): Builder => $query->withTrashed($request->user()->hasPermissionTo("showInactiveUsers")))
```
Comment (krzysztofrewak) [inline app/Http/Controllers/VacationLimitController.php:30]: As this is really common thing in codebase, maybe `$user->canSeeInactiveUsers` would be nice helper?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/VacationLimitController.php (hunk 4 above)
Comment (kamilpiech97) [reply app/Http/Controllers/VacationLimitController.php]: changed

File: app/Http/Controllers/EquipmentController.php (hunk 2 above)
Comment (kamilpiech97) [reply app/Http/Controllers/EquipmentController.php]: changed

File: app/Http/Controllers/EquipmentController.php (hunk 1 above)
Comment (kamilpiech97) [reply app/Http/Controllers/EquipmentController.php]: typed

**Review by krzysztofrewak — APPROVED** [review]

**Review by EwelinaSkrzypacz — APPROVED** [review]

**Pull request merged**